java/server/src/org/openqa/grid/common/RegistrationRequest.java

```diff
@@ -203,6 +203,9 @@ public static RegistrationRequest build(GridNodeConfiguration configuration, Str
     if (pendingConfiguration.port != null) {
       pendingRequest.configuration.port = pendingConfiguration.port;
     }
+    if (pendingConfiguration.remoteHost != null) {
+      pendingRequest.configuration.remoteHost = pendingConfiguration.remoteHost;
+    }
```

Comment:
This is not really needed, the line added in `GridNodeConfiguration.java` is what really fixes the `remoteHost` regression.

Reply:
You're right :) corrected: https://github.com/SeleniumHQ/selenium/pull/6627/commits/91e71499f37dc1852db0df9bbbb7480acbf0cef2